src/components/Workflow/SearchableWorkflowNameList.tsx

```diff
@@ -11,27 +11,93 @@ import { WorkflowExecutionPhase } from 'models/Execution/enums';
 import { Shimmer } from 'components/common/Shimmer';
 import { WorkflowExecutionIdentifier } from 'models/Execution/types';
 import { debounce } from 'lodash';
-import { Typography } from '@material-ui/core';
+import {
+  IconButton,
+  Typography,
+  FormControlLabel,
+  Checkbox,
+  FormGroup,
+  Button,
+  CircularProgress,
+} from '@material-ui/core';
+import UnarchiveOutline from '@material-ui/icons/UnarchiveOutlined';
+import ArchiveOutlined from '@material-ui/icons/ArchiveOutlined';
+import { useMutation } from 'react-query';
+import { WorkflowExecutionState } from 'models/Workflow/enums';
+import { updateWorkflowState } from 'models/Workflow/api';
+import { useState } from 'react';
+import { useSnackbar } from 'notistack';
 import { WorkflowListStructureItem } from './types';
 import ProjectStatusBar from '../Project/ProjectStatusBar';
 import { workflowNoInputsString } from '../Launch/LaunchForm/constants';
 import { SearchableInput } from '../common/SearchableList';
 import { useSearchableListState } from '../common/useSearchableListState';
 import { useWorkflowInfoItem } from './useWorkflowInfoItem';
+import t from '../Executions/Tables/WorkflowExecutionTable/strings';
+import { getArchiveStateString, isWorkflowArchived } from './utils';
 
 interface SearchableWorkflowNameItemProps {
   item: WorkflowListStructureItem;
 }
 
+interface SearchableWorkflowNameItemActionsProps {
+  item: WorkflowListStructureItem;
+  setHideItem: (hide: boolean) => void;
+}
+
 interface SearchableWorkflowNameListProps {
   workflows: WorkflowListStructureItem[];
+  onArchiveFilterChange: (showArchievedItems: boolean) => void;
+  showArchived: boolean;
 }
 
+export const showOnHoverClass = 'showOnHover';
+
 const useStyles = makeStyles(() => ({
+  actionContainer: {
+    display: 'block',
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    height: '100%',
+  },
+  actionProgress: {
+    width: '100px',
+    textAlign: 'center',
+    top: '42%',
+    display: 'block',
+    position: 'absolute',
+    right: 0,
+  },
+  archiveButton: {
+    right: '30px',
+    position: 'relative',
+    top: '42%',
+    height: 'auto',
+  },
+  archiveCheckbox: {
+    whiteSpace: 'nowrap',
+  },
+  confirmationBox: {
+    height: '100%',
+    [`& > button`]: {
+      height: '100%',
+    },
+  },
+  confirmationButton: {
+    borderRadius: 0,
+    minWidth: '100px',
+    minHeight: '53px',
+  },
   container: {
     padding: 13,
     paddingRight: 71,
   },
+  filterGroup: {
+    display: 'flex',
```

Comment:
nit: Also as you are already here  😇 maybe you will change paddings / borderRadius and margins to use theme.spacing() values. Value 1 corresponds to 8px. we can use float numbers as 1.5 or 1.25 - but ideally we should stick to whole numbers. It will allow us to standardize our styling in future. I tried next values here:
```
  container: {
    padding: theme.spacing(2),
    paddingRight: theme.spacing(5),
  },
  filterGroup: {
    display: 'flex',
    flexWrap: 'nowrap',
    flexDirection: 'row',
    margin: theme.spacing(4, 5, 2, 2),
  },
  searchInputContainer: {
    paddingLeft: 0,
  },
```